CentralAuthPlugin.php

```diff
@@ -225,21 +226,30 @@ public function filterLoginAdapter($adapter, $args)
                 }
             }
 
+            // Get LDAP bind password from the ini file if configuration
+            // is set up to do this and a bind username has been entered
+            if (isset($options['username']) && $options['use_ini']) {
+                $ldap_ini = parse_ini_file ('ldap.ini', true);
+                if (isset($ldap_ini['ldap']['bind_password'])) {
+                    $options['password'] = $ldap_ini['ldap']['bind_password'];
+                }
+            }
+
+            // Remove $options['use_ini'] because it's not an option that
+            // is recognized by Zend_Ldap, it's only use is to tell us
+            // where to find the 'password' option
+            if (isset($options['use_ini'])) {
+                unset($options['use_ini']);
+            }
+
             // Create new auth adapter with the options, username and password.
             $adapterLdap = new CentralAuth_LdapAdapter(
                 array('ldap' => $options),
                 $form->getValue('username'),
                 $form->getValue('password')
             );
 
-            // Attempt to authenticate using LDAP.
-            $result = $adapterLdap->authenticate();
-
-            // If the user authenticated or LDAP authentication is required,
-            // return the LDAP auth adapter as the version to use.
-            if ($result->isValid() || $ldap == 'required') {
-                return $adapterLdap;
```

Comment:
@kloor This function actually returns an LDAP adapter to the Omeka authentication adapter filter, so the Omeka login system will use the Ldap adapter that was returned by the filter and authenticate with the LDAP server a second time. As a reminder, apparently in the Zend code, it authenticates twice with the ldap server if you use a bind with a username and password. This results in a total of 4 calls to the ldap server for my particular situation.